src/Factory/Product/ProductVariantViewFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusLagersystemPlugin\Factory\Product;
+
+use Setono\SyliusLagersystemPlugin\View\Product\ProductVariantView;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+
+final class ProductVariantViewFactory implements ProductVariantViewFactoryInterface
+{
+    /** @var string */
+    private $productVariantViewClass;
+
+    public function __construct(string $productVariantViewClass)
+    {
+        $this->productVariantViewClass = $productVariantViewClass;
+    }
+
+    public function create(ProductVariantInterface $variant, string $localeCode): ProductVariantView
+    {
+        $translation = $variant->getTranslation($localeCode);
+
+        /** @var ProductVariantView $variantView */
+        $variantView = new $this->productVariantViewClass();
+        $variantView->id = $variant->getId();
+        $variantView->code = $variant->getCode();
+        $variantView->name = $translation->getName();
+        $variantView->weight = (int) ceil(1000 * $variant->getShippingWeight());
```

Comment:
I guess it isn't defined anywhere that weight in Sylius is kilos. I think it would make more sense that weight in Sylius would be grams just like money is cents. Don't you think this should be our point of view?

No matter what I think you should add a note to the readme

Reply:
The weight in Sylius is float, so I think that means its about kilos.
Anyway, its a great idea to add a hint at README about that, thanks.